### PR TITLE
Create articles

### PR DESCRIPTION
**Pull request by holgertidemand**
#### Description
Changes proposed in this pull request:

* Adds form for creating articles
* Adds flash message for successfully & unsuccessfully created articles 
#### What I have learned working on this feature:
* Better understanding for the from_with helper in rails.

### CODE REVIEW

File: features/step_definitions/basic_steps.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Then("I should see {string}") do |message|
+  page.should have_content message
+end
+
+def page_path(path)
+  site_path =  if path == 'Create Article'
+                new_article_path
+               end
+  site_path
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: ???

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Then("I fill in {string} with {string}") do |element, content|
+  fill_in(element, :with => content)
+end
+
+Then("I click on {string}") do |button_name|
+  click_link_or_button(button_name)
+end
+
+Then("I should see {string}") do |message|
+  page.should have_content message
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Please use expect to make assertions, as we’ve done through the entire course

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Given("I am on the {string} Page") do |page_name|
+  visit page_path(page_name)
+end
+
+Then("I fill in {string} with {string}") do |element, content|
+  fill_in(element, :with => content)
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Please use new hash syntax

File: app/controllers/articles_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+class ArticlesController < ApplicationController
+
+  def create
+    @article = Article.new(article_params)
+    if @article.save
+      flash[:success] = 'Article successfully created'
+      redirect_to article_path(@article)
+    else
+      if params[:title].blank? || params[:body].blank?
```
Comment (tochman) [inline app/controllers/articles_controller.rb]: This one is a little bit problematic for me.  Please use the error message active record provides for us if article is not being saved.

File: db/migrate/20180314143525_create_active_storage_tables.active_storage.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      t.string     :name,     null: false
+      t.references :record,   null: false, polymorphic: true, index: false
+      t.references :blob,     null: false
+
+      t.datetime :created_at, null: false
+
+      t.index [ :record_type, :record_id, :name, :blob_id ], name: "index_active_storage_attachments_uniqueness", unique: true
+    end
+  end
+end
```
Comment (tochman) [inline db/migrate/20180314143525_create_active_storage_tables.active_storage.rb]: If you do not use active storage in this pull request, please  do not create the migrations

**Review by tochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/basic_steps.rb (hunk 1 above)
Comment (holgertidemand) [reply features/step_definitions/basic_steps.rb]: Raoul showed this to us. It's a function that lets us reuse steps more than one time. In this case this is all it needs to do. But when we will test other paths we will add them to the if statement.

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (holgertidemand) [reply features/step_definitions/basic_steps.rb]: Will fix that. 
We forgot and looked in the capybara documentation.

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (holgertidemand) [reply features/step_definitions/basic_steps.rb]: Got it! 
We forgot and looked in the capybara documentation.

File: db/migrate/20180314143525_create_active_storage_tables.active_storage.rb (hunk 5 above)
Comment (holgertidemand) [reply db/migrate/20180314143525_create_active_storage_tables.active_storage.rb]: Yes, totally get that! Missed it when i went through the PR myself.

File: app/controllers/articles_controller.rb (hunk 4 above)
Comment (holgertidemand) [reply app/controllers/articles_controller.rb]: Will have to look in to this. But we can make that happen.

File: features/step_definitions/basic_steps.rb (hunk 1 above)
Comment (tochman) [reply features/step_definitions/basic_steps.rb]: @diraulo Is this your implementation?

File: app/controllers/articles_controller.rb [hunk 6]
```diff
@@ -18,4 +16,13 @@ def create
   def article_params
     params.require(:article).permit(:title, :body)
   end
+
+
+  def error_message(article)
+    output = ''
+    article.errors.full_messages.each do |error|
+      output += "#{error} "
```
Comment (tochman) [inline app/controllers/articles_controller.rb:24]: Please take a second look at this and review it in the browser. It is not a blocker imo but will this not mean that the error messages (if there are more than one) end up as one single string without spaces between them?

**Review by tochman — APPROVED** [review]

File: app/controllers/articles_controller.rb (hunk 6 above)
Comment (holgertidemand) [reply app/controllers/articles_controller.rb:24]: It will end up as one string but with a space between them! 
We didn't really know how to make use of multiple flashes with out making the first one disappear.
Would love some pointers! 
 
We decided to go with it for now and when we get to the styling we would make it look better.

**Comment by larathordar** [comment]
Looks good !

File: features/step_definitions/basic_steps.rb (hunk 1 above)
Comment (diraulo) [reply features/step_definitions/basic_steps.rb]: @holgertidemand For future reference, you should learn to own the code you produce. The moment you get a code from a coach or someone else it becomes yours and you need to be able to defend your decision to use it without pointing at someone else. 

Here's what I showed you back then https://github.com/CraftAcademy/evensteven/pull/10#discussion_r171255271

Your team ended up going with the above implementation or something similar which I let through for various reasons related to the context in which you were working on. Now that you know better, this was not going to get an approval from me either.

Bottom line here is *own the code you produce* - You will not tell your boss tomorrow in a code review _I got this from StackOverflow, and I just used it..._ without thinking about its impact on the application.

Comment (holgertidemand) [reply features/step_definitions/basic_steps.rb]: Copy that! @diraulo 
I will OWN my code! :D
